kernel/kernel-api/src/test/scala/io/delta/kernel/internal/snapshot/LogSegmentSuite.scala

```diff
@@ -40,74 +40,114 @@ class LogSegmentSuite extends AnyFunSuite with MockFileSystemClientUtils {
   }
 
   test("constructor -- valid case (non-empty)") {
-    new LogSegment(logPath, 12, deltasFs11To12List, checkpointFs10List, 1)
+    new LogSegment(logPath, 12, deltasFs11To12List, Collections.emptyList(), checkpointFs10List, 1)
   }
 
   test("constructor -- null arguments => throw") {
     // logPath is null
     intercept[NullPointerException] {
-      new LogSegment(null, 1, Collections.emptyList(), Collections.emptyList(), -1)
+      new LogSegment(
```

Comment:
do you think it's worth (not in this PR) creating a LogSegmentBuilder so we don't have to keep changing the constructor every time we add a new file type?

I think @huan233usc will be add another one for CRC files ...

Reply:
Yeah I think a builder would help a lot with the number of Collections.emptyList(), especially when Xin rebases the CRC PR on top of this ...

And probably not a "test-only" one? (Or, is that a common convention?) Probably just right in `LogSegment.java` but perhaps we could use `@VisibleForTesting` so that our prod code doesn't use it (though, it could?) i.e. it is only there for the test code to use?

Reply:
Not sure. But you could code a "Builder" that's only in test code, if we didn't actually want to use it in prod. Like why would you put it in prod code if you don't want to use it? (I guess maybe it needs access to internal details so has to be within the class or something, not sure if that would apply here)